<launch>
    <!-- <arg name="map_file" default="$(find course_agv_gazebo)/config/map.yaml"/> -->

	<node pkg="course_agv_ekf" type="icp_ekf.py" name="localization" output="screen" >
		<param name="range_threshold" value="0.5"/>
		<param name="radius_max_th" value="0.4"/>
		<param name="landMark_min_pt" value="2"/>
		<param name="min_match" value="2"/>

		<!--icp params-->
		<param name="robot_x" value="0"/>
		<param name="robot_y" value="0"/>
		<param name="robot_theta" value="0"/>

		<param name="max_iter" value="30"/>
		<param name="tolerance" value="0"/>
		<param name="dis_th" value="0.05"/>
		<param name="corner_num" value="150"/>
		<param name="past_frame_num" value='0'/>
	</node>

	<!-- <node pkg = 'course_agv_ekf' type="icp" name='icp' output='screen'> -->
	<!--from a rosbag-->
		<!-- <param name="robot_x" value="0"/>
		<param name="robot_y" value="0"/>
		<param name="robot_theta" value="0"/>

		<param name="max_iter" value="100"/>
		<param name="tolerance" value="0"/>
		<param name="dis_th" value="0.05"/>
	</node> -->

	<node pkg="course_agv_ekf" type="tf" name="my_tf"/>
	
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find course_agv_ekf)/launch/course_agv.rviz" />
</launch>


